execstats: add cumulative contention time to the TraceAnalyzer

asubiotto opened the pull request:
Depends on #59132 
Closes #56797 

This commit adds the summing of each processor's contention time to the
TraceAnalyzer. This stat is returned as a node and query level stat.

Additionally, this commit adds a ContentionTime field to the
StatementStatistics protobuf used in the DB Console.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59299)
<!-- Reviewable:end -->

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59299#-:-MRgFzBaCsFJBdd-3OcQ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Commit:
execstats: add cumulative contention time to the TraceAnalyzer

This commit adds the summing of each processor's contention time to the
TraceAnalyzer. This stat is returned as a node and query level stat.

Additionally, this commit adds a ContentionTime field to the
StatementStatistics protobuf used in the DB Console.

Release note: None

asubiotto commented:
bors r=RaduBerinde
